File: FrontEnd/CareerLink1/src/app/models/Reclamation.ts

import { TypeReclamation } from "./typeReclamation";
import {Reponse } from "./Reponse";
import { Expense } from "./Expense";

export class Reclamation {

    idreclamation!:number;
    datereclamation!: Date;
    description!: string;
 
    typeReclamation!:TypeReclamation;
   importance!:number;
     
 
    reponse!: Reponse[]; 
    expense: Expense = new Expense(); 

   
    
 }

File: BackEnd/CareerLink/src/main/java/tn/esprit/careerlink/entities/Reclamation.java
package tn.esprit.careerlink.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.io.Serializable;
import java.time.LocalDate;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Reclamation implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer idreclamation;
    LocalDate datereclamation;
    String description;
    @Enumerated(EnumType.STRING)
    private StatuReclamation statuReclamation;
    @Enumerated(EnumType.STRING)
    private TypeReclamation typeReclamation;
    @JsonIgnore
    @OneToOne
    Reponse reponse;



}
